lib/python/rsdb.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+
+from ctypes import *
+import os
+
+sled = CDLL("./libsled.so")
+
+sled.sled_create_config.argtypes = ()
+sled.sled_create_config.restype = c_void_p
+
+sled.sled_config_set_path.argtypes = (c_void_p, c_char_p)
+
+sled.sled_free_config.argtypes = (c_void_p,)
+
+sled.sled_open_tree.argtypes = (c_void_p,)
+sled.sled_open_tree.restype = c_void_p
+
+sled.sled_free_tree.argtypes = (c_void_p,)
+
+sled.sled_get.argtypes = (c_void_p, c_char_p, c_size_t, POINTER(c_size_t))
+sled.sled_get.restype = c_char_p
+
+sled.sled_scan.argtypes = (c_void_p, c_char_p, c_size_t, POINTER(c_size_t))
+sled.sled_scan.restype = c_void_p
+
+sled.sled_set.argtypes = (c_void_p, c_char_p, c_size_t, c_char_p, c_size_t)
+sled.sled_set.restype = None
+
+sled.sled_del.argtypes = (c_void_p, c_char_p, c_size_t)
+sled.sled_del.restype = None
```

Comment:
from python, we need to describe the type signatures in this way for `ctypes` to properly transform data before calling the c-compatible library. this will later probably use one of the other FFI libraries to work in a more flexible way, but for now it's cool to be calling this from python :)